recipes/repository.rb

```diff
@@ -26,17 +26,25 @@
     action :install
   end
 
-  uri = node['datadog']['agent6'] ? node['datadog']['agent6_aptrepo'] : node['datadog']['aptrepo']
-  distribution = node['datadog']['agent6'] ? node['datadog']['agent6_aptrepo_dist'] : node['datadog']['aptrepo_dist']
-  components = node['datadog']['agent6'] ? ['main', '6'] : ['main']
+  case node['datadog']['agent_major_version'].to_i
+  when 7
+    components = ['7']
+  when 6
+    components = ['main', '6'] # early versions of A6 were deployed to main. TODO: remove main if 6 contains all versions
```

Comment:
Checked this: we started releasing Agent 6 to the `6` component starting with `6.0.0-beta.9`. Earlier betas were only released on the `main` component. See:

* [Packages file on the `beta` distribution of `6`](https://s3.amazonaws.com/apt.datadoghq.com/dists/beta/6/binary-x86_64/Packages)
* [Packages file on the `beta` distribution of `main`](https://s3.amazonaws.com/apt.datadoghq.com/dists/beta/main/binary-x86_64/Packages)

At the moment almost nobody uses 6.0.0 betas that are earlier than `beta.9`, so I think it's safe to drop `main` here. Ping me on chat if you want more precise usage stats on early 6.0.0 betas :)